### PR TITLE
BUGZ-812 - Log system info on domain connect or connect refusal

### PR DESCRIPTION
**Pull request by roxanneskelly**
https://highfidelity.atlassian.net/browse/BUGZ-812

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/20105/HighFidelity-Beta-PR15828-e39ccff.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/20105/HighFidelity-Quest-Beta-PR15828-e39ccff.apk)

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/20105/HighFidelity-Beta-PR15828-e39ccff.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/20105/HighFidelity-Beta-Interface-PR15828-e39ccff.exe

File: domain-server/src/NodeConnectionData.cpp [hunk 1]
```diff
@@ -37,7 +37,11 @@ NodeConnectionData NodeConnectionData::fromDataStream(QDataStream& dataStream, c
         dataStream >> newHeader.machineFingerprint;
 
         // and the operating system type
-        dataStream >> newHeader.SystemInfo;
+        QByteArray compressedSystemInfo;
+        dataStream >> compressedSystemInfo;
+        if(!compressedSystemInfo.isEmpty()) {
```
Comment (howard-stearns) [inline domain-server/src/NodeConnectionData.cpp]: coding standard - space after if

File: libraries/networking/src/NodeList.cpp [hunk 2]
```diff
@@ -418,19 +421,20 @@ void NodeList::sendDomainServerCheckIn() {
...
-#else
-            systemInfo = "OS:Unknown";
-#endif
-            packetStream << systemInfo;
+            auto desc = platform::getAll();
+
+            QByteArray systemInfo(desc.dump().c_str());
+            QByteArray compressedSystemInfo = qCompress(systemInfo);
+
+            if(compressedSystemInfo.size() > MAX_SYSTEM_INFO_SIZE) {
```
Comment (howard-stearns) [inline libraries/networking/src/NodeList.cpp]: coding standard - space after if

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/20110/HighFidelity-Beta-PR15828-3c0c141.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/20110/HighFidelity-Beta-Interface-PR15828-3c0c141.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/20110/HighFidelity-Beta-PR15828-3c0c141.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/20110/HighFidelity-Beta-Interface-PR15828-3c0c141.dmg

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/20110/HighFidelity-Beta-PR15828-3c0c141.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/20110/HighFidelity-Quest-Beta-PR15828-3c0c141.apk)

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/20114/HighFidelity-Beta-PR15828-c5883a8.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/20114/HighFidelity-Beta-Interface-PR15828-c5883a8.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/20114/HighFidelity-Beta-PR15828-c5883a8.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/20114/HighFidelity-Beta-Interface-PR15828-c5883a8.dmg

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/20114/HighFidelity-Beta-PR15828-c5883a8.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/20114/HighFidelity-Quest-Beta-PR15828-c5883a8.apk)

**Pull request merged**